### PR TITLE
Feature/4 integrate redux state management

### LINKED ISSUE #4: Integrate Redux state management

### LINKED ISSUE #23: Security vulnerabilities due to downgrade to react 17.0.2
To resolve peer dependencies of many of the libraries we use, we had to downgrade from react v18.0.0 to react v17.0.2 in #22 . The audit process reveals that this results in 19 new high security vulnerabilities. 

We need to discuss on how to resolve these.

### PR DESCRIPTION
**Pull request by ValiHar**
Integrated following components from @n-krueger PoC: 
- Redux
- Services
- Prettier
- Environments
- Storybooks

Furthermore, I downgraded the project to react 17.0.2 due to peer dependency issues

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -72,7 +72,7 @@ typings/
 .yarn-integrity
 
 # dotenv environment variables file
-.env
+env/.env
```
Comment (n-krueger) [inline .gitignore:75]: Does this still work for you if it is stored in the subdirectory?

**Review by n-krueger — APPROVED** [review]

**Review by n-krueger — CHANGES_REQUESTED** [review]
Please fix the tests

File: .gitignore (hunk 1 above)
Comment (ValiHar) [reply .gitignore:75]: It does if you explicitly specify the path with the `--env-file` flag :) If that's too cumbersome we can move the files back.

**Comment by ValiHar** [comment]
> Please fix the tests

@n-krueger Are you talking about the test for the App component? Because that component is removed in #17

**Comment by n-krueger** [comment]
> @n-krueger Are you talking about the test for the App component? Because that component is removed in https://github.com/FHIR-Vaccination-Pass/patient-dashboard/issues/17

Yes, if you want to merge #17 all at once then I would suggest we close this PR. :)

File: .gitignore (hunk 1 above)
Comment (n-krueger) [reply .gitignore:75]: Too cumbersome IMO

**Comment by ValiHar** [comment]
Alright, then we'll merge it with #17

**Pull request closed without merging**